test/alarm.c

```diff
@@ -0,0 +1,19 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <signal.h>
+
+void
+handle_alarm(int sig)
+{
+  exit(EXIT_SUCCESS);
+}
+
+int
+main()
+{
+  signal(SIGALRM, handle_alarm);
+  alarm(15);
+  while (1);
```

Comment:
Sorry for not noticing it sooner. Can you change it to `while (1) sleep(1);`? This would avoid cpu spinning.